src/sqlstore/timeline.py

```diff
@@ -0,0 +1,234 @@
+    platformId = Column(String(7), primary_key=True)
+    gameId = Column(BigInteger, primary_key=True)
+    frameId = Column(Integer, primary_key=True)
+    killId = Column(Integer, Identity(always=True), primary_key=True)
+    # eventId = Column(Integer, ForeignKey("match_timeline_event.eventId"))   # TODO: is this correct?
+    assistingParticipantIds = Column(PickleType)    # serialized list of participant ids
+    bounty = Column(Integer)
+    killStreakLength = Column(Integer)
+    killerId = Column(Integer)
+    position_x = Column(Integer)
+    position_y = Column(Integer)
+    shutdownBounty = Column(Integer)
+    timestamp = Column(Integer)
+    type = Column(String(30))
+    victimId = Column(Integer)
+    timeCreated = Column("timeCreated", DateTime(timezone=True), server_default=func.now())
+    lastUpdate = Column("lastUpdate", DateTime(timezone=True), onupdate=func.now())
+
+    def __init__(self, **kwargs):
+        for attr in ('platformId', 'gameId', 'frameId', 'killId', 'eventId', 'assistingParticipantIds', 'bounty',
+                     'killStreakLength', 'killerId', 'position_x', 'position_y', 'shutdownBounty', 'timestamp'):
+            setattr(self, attr, kwargs.get(attr))
+
+    def __repr__(self):
+        return f"{self.platformId}_{self.gameId} at frame {self.frameId} (id: {self.killId}) {self.killerId} killed " \
+               f"{self.victimId}"
+
+
+class SQLTimelineVictimDamageDealt(Base):
+    __tablename__ = "match_timeline_kill_victimdmgdealt"
+    platformId = Column(String(7), primary_key=True)    # TODO: these 4 rows should be foreign, not primary keys?
+    gameId = Column(BigInteger, primary_key=True)
+    frameId = Column(Integer, primary_key=True)
+    damageId = Column(Integer, Identity(always=True), primary_key=True)
+    #killId = Column(Integer, ForeignKey("match_timeline_event_kill.killId"), nullable=False)
+    basic = Column(Boolean)
+    magicDamage = Column(Integer)
+    name = Column(String(30))
+    participantId = Column(Integer)
+    physicalDamage = Column(Integer)
+    spellName = Column(String(70))
+    spellSlot = Column(Integer)
+    trueDamage = Column(Integer)
+    type = Column(String(40))
+    timeCreated = Column("timeCreated", DateTime(timezone=True), server_default=func.now())
+    lastUpdate = Column("lastUpdate", DateTime(timezone=True), onupdate=func.now())
+
+    def __init__(self, **kwargs):
+        for attr in ('platformId', 'gameId', 'frameId', 'damageId', 'killId', 'magicDamage', 'name', 'participantId',
+                     'physicalDamage', 'spellName', 'spellSlot', 'trueDamage', 'type'):
+            setattr(self, attr, kwargs.get(attr))
+
+    def __repr__(self):
+        return f"{self.platformId}_{self.gameId} at frame {self.frameId} (id: {self.damageId}) {self.name} " \
+               f"dealt damage before being killed"
+
+
+class SQLTimelineVictimDamageReceived(Base):    # TODO: can/should this table be merged with victimdmgdealt?
```

Comment:
Maybe merge them. Both have really similar classes.

Reply:
my idea was to distinguish between them by a boolean variable indicating which is damage received and dealt